rpc/dialer_test.go

```diff
@@ -357,3 +358,146 @@ func TestWithDialStatsHandler(t *testing.T) {
 
 	test.That(t, stats.clientConnections, test.ShouldBeGreaterThan, 1)
 }
+
+func TestWithUnaryInterceptor(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	rpcServer, err := NewServer(
+		logger,
+		WithAuthHandler("fake", MakeFuncAuthHandler(func(ctx context.Context, entity, payload string) (map[string]string, error) {
+			if entity != "passmethrough" {
+				return nil, errors.New("nope")
+			}
+			return map[string]string{}, nil
+		}, func(ctx context.Context, entity string) (interface{}, error) {
+			return entity, nil
+		})),
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	httpListener, err := net.Listen("tcp", "localhost:0")
+	test.That(t, err, test.ShouldBeNil)
+
+	err = rpcServer.RegisterServiceServer(
+		context.Background(),
+		&pb.EchoService_ServiceDesc,
+		&echoserver.Server{},
+		pb.RegisterEchoServiceHandlerFromEndpoint,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	go func() {
+		rpcServer.Serve(httpListener)
+	}()
+
+	var interceptedMethods []string
+	fakeInterceptor := func(
+		ctx context.Context,
+		method string,
+		req, reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) error {
+		interceptedMethods = append(interceptedMethods, method)
+		return invoker(ctx, method, req, reply, cc, opts...)
+	}
+	conn, err := Dial(
+		context.Background(),
+		httpListener.Addr().String(),
+		logger,
+		WithUnaryClientInterceptor(fakeInterceptor),
+		WithDialDebug(),
+		WithInsecure(),
+		WithEntityCredentials("passmethrough", Credentials{Type: "fake"}),
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	client := pb.NewEchoServiceClient(conn)
+	_, err = client.Echo(context.Background(), &pb.EchoRequest{Message: "hello"})
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, conn.Close(), test.ShouldBeNil)
+	test.That(t, rpcServer.Stop(), test.ShouldBeNil)
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(
+		t,
+		interceptedMethods,
+		test.ShouldResemble,
+		[]string{
+			"/proto.rpc.webrtc.v1.SignalingService/OptionalWebRTCConfig",
+			"/proto.rpc.v1.AuthService/Authenticate",
+			"/proto.rpc.examples.echo.v1.EchoService/Echo",
+			"/proto.rpc.v1.AuthService/Authenticate",
```

Comment:
we probably should exempt auth and signaling method like we do for server-side interceptors

Reply:
@edaniels any thoughts on this?